sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -537,6 +541,7 @@ class Analyzer(
     def containsAggregates(exprs: Seq[Expression]): Boolean = {
       exprs.foreach(_.foreach {
         case agg: AggregateExpression => return true
```

Comment:
if we are not going to be removing the old aggregate expression for a while, how about creating a new aggregateexpression that's the shared parent, and rename existing one aggregateexpression1? i worry we miss places that we match on.